### PR TITLE
chore(build-definitions): update owners

### PR DESCRIPTION
**Pull request by MartinBasti**
Build-definitions repository is mainatained by multiple teams and those teams members should be also approvers for build-pipeline-selector to reduce load on the current approvers.

Approve plugin doesn't currently support filters, so the only way how to achieve this was to make a subdirectory.
https://www.kubernetes.dev/docs/guide/owners/

### CODE REVIEW

**Comment by MartinBasti** [comment]
I have to update [build-definitions tekton task](https://github.com/redhat-appstudio/build-definitions/blob/main/.tekton/push.yaml#L99)   as well, so please let me know if you agree with naming of subdirectory

**Comment by mmorhun** [comment]
To me `build-definitions` doesn't reflect the selection of the build pipeline. Also, it confuses with the `build-definitions` repository. What about `build-pipeline-selectors` directory name?

**Comment by MartinBasti** [comment]
updated

File: components/build-service/base/build-pipeline-selectors/OWNERS [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- tisutisu
+- chmeliik
+- dirgim
+- hongweiliu17
+- jsztuka
+- jinqi7
+- sonam1412
+- MartinBasti
+- Josh-Everett
+- 14rcole
```
Comment (sonam1412) [inline components/build-service/base/build-pipeline-selectors/OWNERS:17]: Should we please also add Dheeraj in the list. Github user account: dheerajodha

File: components/build-service/base/build-pipeline-selectors/OWNERS [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (sonam1412) [inline components/build-service/base/build-pipeline-selectors/OWNERS:34]: Should we please also add Dheeraj in the list. Github user account: dheerajodha

**Review by mmorhun — APPROVED** [review]

File: components/build-service/base/build-pipeline-selectors/OWNERS (hunk 1 above)
Comment (MartinBasti) [reply components/build-service/base/build-pipeline-selectors/OWNERS:17]: added

**Comment by MartinBasti** [comment]
Follow up PR: https://github.com/redhat-appstudio/build-definitions/pull/608

Please `/lgtm` this

**Review by sonam1412 — APPROVED** [review]
LGTM

**Pull request merged**